great_expectations/datasource/data_connector/configured_asset_sql_data_connector.py

```diff
@@ -108,21 +113,43 @@ def _get_batch_identifiers_list_from_data_asset_config(
             table_name = data_asset_name
 
         if "splitter_method" in data_asset_config:
-            splitter_fn = getattr(self, data_asset_config["splitter_method"])
-            split_query = splitter_fn(
-                table_name=table_name, **data_asset_config["splitter_kwargs"]
-            )
+            SPLITTER_METHOD_TO_GET_UNIQUE_BATCH_IDENTIFIERS_MAPPING: dict = {
```

Comment:
@anthonyburdi do we really want to define a constant this way?..

Reply:
So I thought a lot about this and didn't come up with a better solution. Basically this is a placeholder to move the new splitter methods to the execution engine until we can do a better refactoring of the splitter handling. 

Do you mean because it is in all caps, or that it is a dict or that it is defined within the method (I wanted it to not be accessible outside this method) or that the type hint could be better?

Reply:
@anthonyburdi because it is all capitals -- which, presumably by convention, implies a constant.  We can avoid this "abuse of style" if we wanted to.  Happy to discuss.

Reply:
Moved this to a class level constant and also split up `_get_batch_identifiers_list_from_data_asset_config()`. Hope you like it! There is more refactor we can do in future PRs, some of which is noted in docstrings.